### PR TITLE
DOC: Documentation for pandas.core.indexes.api

### PR DESCRIPTION
**Pull request by lowerthansound**
The functions on pandas.core.indexes.api had almost no documentation. This is an attempt to explain them better and a progress towards #22915.

### CODE REVIEW

File: pandas/core/indexes/api.py [hunk 1]
```diff
@@ -44,16 +44,20 @@
 
 
 def _get_objs_combined_axis(objs, intersect=False, axis=0, sort=True):
-    # Extract combined index: return intersection or union (depending on the
-    # value of "intersect") of indexes on given axis, or None if all objects
-    # lack indexes (e.g. they are numpy arrays)
+    """Extract combined index: return intersection or union (depending on the
+    value of "intersect") of indexes on given axis, or None if all objects
+    lack indexes (e.g. they are numpy arrays)
```
Comment (jreback) [inline pandas/core/indexes/api.py]: can u add Parameters and Returns to each of these

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/api.py [hunk 2]
```diff
@@ -44,16 +44,50 @@
...
     if obs_idxes:
         return _get_combined_index(obs_idxes, intersect=intersect, sort=sort)
 
 
 def _get_combined_index(indexes, intersect=False, sort=False):
+    """Return the union or intersection of indexes
+
+    Parameters
+    ----------
+    indexes: a list of Index or array-like objects
```
Comment (araraonline) [inline pandas/core/indexes/api.py]: Actually, this should be 'a list of Index or *list*objects'

Will fix soon

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
context looks ok, @datapythonista or @jorisvandenbossche if you want to glance at the doc-strings; these are internal only

**Review by datapythonista — CHANGES_REQUESTED** [review]
Looks good, it's nice to have docs in those functions.

Few comments about formatting that apply to all docstrings:
- Start he docstring in the next line of `"""`, not in the same
- Missing space before colon in Parameters `param_name : type`
- Missing period at the end of the parameter descriptions, and the returns description.
- Use Python types for parameter types (or returns), for example `bool` instead of `boolean`. And have just the type, for example `list of bool` instead of `a list of bool`

File: pandas/core/indexes/api.py [hunk 3]
```diff
@@ -126,6 +191,26 @@ def conv(i):
...
 
 def _sanitize_and_check(indexes):
+    """
+    Verify the type of indexes and convert lists to Index.
+
+    Cases:
+
+    - [list, list, ...]: Return ([list, list, ...], 'list')
+    - [list, Index, ...]: Return _sanitize_and_check([Index, Index, ...])
+        Lists are sorted and converted to Index
```
Comment (araraonline) [inline pandas/core/indexes/api.py]: Will add period to this too

File: pandas/core/indexes/api.py [hunk 4]
```diff
@@ -144,6 +229,21 @@ def _sanitize_and_check(indexes):
 
 
 def _get_consensus_names(indexes):
+    """
+    Give a consensus 'names' to indexes,
```
Comment (araraonline) [inline pandas/core/indexes/api.py]: my bad

**Review by datapythonista — APPROVED** [review]
Thanks for adding these docs, and sorry for the delay in reviewing.

**Pull request merged**